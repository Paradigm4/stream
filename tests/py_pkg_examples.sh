#!/bin/bash

set -o errexit

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"

if [ "$1" != "" ]
then
    PRE="$1"
fi


# 1.
$PRE iquery --afl --query "
    store(
      apply(
        build(<x:int64 not null>[i=1:10:0:5], i),
        y, double(i) * 10 + .1,
        z, 'foo' + string(i)),
      foo)" \
>  $DIR/py_pkg_examples.out

$PRE iquery --afl --query "
    stream(
      foo,
      'python -u /stream/py_pkg/examples/1-map-finalize.py',
      'format=feather',
      'types=int64,double,string',
      'names=x,y,info')" \
>> $DIR/py_pkg_examples.out

$PRE iquery --afl --query "remove(foo)"


# 2.
python $DIR/../py_pkg/examples/2-pack-func.py \
>> $DIR/py_pkg_examples.out


# 3.
$PRE iquery --afl --query "
    store(
      apply(
        build(<x:int64 not null>[i=1:10:0:5], i),
        y, double(i) * 10 + .1,
        z, 'foo' + string(i)),
      foo)" \
>> $DIR/py_pkg_examples.out

$PRE iquery --afl --query "
    stream(
      foo,
      'python -u /stream/py_pkg/examples/3-read-write.py',
      'format=feather',
      'types=int64,double,string')" \
>> $DIR/py_pkg_examples.out

$PRE iquery --afl --query "remove(foo)"


# 4.
python $DIR/../py_pkg/examples/4-machine-learning.py \
>> $DIR/py_pkg_examples.out


# Diff
diff --ignore-trailing-space \
     $DIR/py_pkg_examples.out $DIR/py_pkg_examples.expected
