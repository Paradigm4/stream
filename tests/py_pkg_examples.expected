{instance_id,chunk_no,value_no} count
{0,0,0} 5
{1,0,0} 5
Query was executed successfully
{i} x,y,z
{1} 1,10.1,'foo1'
{2} 2,20.1,'foo2'
{3} 3,30.1,'foo3'
{4} 4,40.1,'foo4'
{5} 5,50.1,'foo5'
{6} 6,60.1,'foo6'
{7} 7,70.1,'foo7'
{8} 8,80.1,'foo8'
{9} 9,90.1,'foo9'
{10} 10,100.1,'foo10'
{instance_id,chunk_no,value_no} x,y,info
{0,0,0} 40,400.5,'local'
{0,1,0} 40,400.5,'total'
{1,0,0} 15,150.5,'local'
{1,1,0} 15,150.5,'total'
[(1, 0, 0, (255, 1.))]
Query was executed successfully
{i} x,y,z
{1} 1,10.1,'foo1'
{2} 2,20.1,'foo2'
{3} 3,30.1,'foo3'
{4} 4,40.1,'foo4'
{5} 5,50.1,'foo5'
{6} 6,60.1,'foo6'
{7} 7,70.1,'foo7'
{8} 8,80.1,'foo8'
{9} 9,90.1,'foo9'
{10} 10,100.1,'foo10'
{instance_id,chunk_no,value_no} a0,a1,a2
{0,0,0} 6,60.1,'foo6'
{0,0,1} 7,70.1,'foo7'
{0,0,2} 8,80.1,'foo8'
{0,0,3} 9,90.1,'foo9'
{0,0,4} 10,100.1,'foo10'
{1,0,0} 1,10.1,'foo1'
{1,0,1} 2,20.1,'foo2'
{1,0,2} 3,30.1,'foo3'
{1,0,3} 4,40.1,'foo4'
{1,0,4} 5,50.1,'foo5'
